flaskr/app.py

```diff
@@ -99,12 +99,17 @@ def calc_weighted_relevancy_rating(self):
             return self.relevance_rating_description
 
     def update_user_score(self):
-        upvotes = [vote for vote in self.user_votes if vote.type == 'up']
-        score_percentage = ((len(upvotes)) / len(self.user_votes))
-        new_weighted_relevancy_rating = (score_percentage * 0.5) + (self.calc_weighted_relevancy_rating() * 0.5)
-        self.weighted_relevancy_rating = new_weighted_relevancy_rating
-        self.user_score = score_percentage
-        db.session.commit()
+        if len(self.user_votes) > 0:
+            upvotes = [vote for vote in self.user_votes if vote.type == 'up']
+            score_percentage = ((len(upvotes)) / len(self.user_votes))
+            new_weighted_relevancy_rating = (score_percentage * 0.5) + (self.calc_weighted_relevancy_rating() * 0.5)
+            self.weighted_relevancy_rating = new_weighted_relevancy_rating
+            self.user_score = score_percentage
+            db.session.commit()
+        else:
+            self.weighted_relevancy_rating = self.calc_weighted_relevancy_rating
+            self.user_score = None
+            db.session.commit()
```

Comment:
(This helps us reset the weighted relevancy when there are no upvotes/downvotes (i.e., we've deleted them)